services/explorer-indexer/src/utils/finder.ts

```diff
@@ -0,0 +1,44 @@
+import { Logger } from '@nestjs/common';
+import TransactionEntity from 'src/modules/transaction/transaction.entity';
+import { Repository } from 'typeorm';
+import util from 'util';
+
+export const safeSaveTx = async (
```

Comment:
I understand why you added this method here, but don't think it is a good place for it. Also the name of the file is not helpful. I think that any changes to the entity should be inside the service which is using the repository. That is more or less the nest way. 
So, I would move this function inside transactions.service and use it from there.